src/components/command-bar/commands/search/unified-search/content.tsx

```diff
@@ -0,0 +1,36 @@
+// Icons
+import { IconDocs16 } from '@hashicorp/flight-icons/svg-react/docs-16'
+import { IconGrid16 } from '@hashicorp/flight-icons/svg-react/grid-16'
+import { IconLearn16 } from '@hashicorp/flight-icons/svg-react/learn-16'
+import { IconPipeline16 } from '@hashicorp/flight-icons/svg-react/pipeline-16'
+// Types
+import type { ReactElement } from 'react'
+import type { UnifiedSearchableContentType } from './types'
+
+/**
+ * Basic heading and icon content for each tab, which we'll build on.
+ */
+export const tabContentByType: Record<
+	UnifiedSearchableContentType,
+	{
+		heading: string
+		icon: ReactElement<React.JSX.IntrinsicElements['svg']>
+	}
+> = {
```

Comment:
💭 What do you think about replacing the the type annotation (`: Record<...`) here with `as const`, (`export const tabContentType = {...} as const`)?

Thought: This frees us up from manual type annotation and lets us lean into automatic type inference instead!

Reply:
If I'm following, i think one downside of `export const tabContentType = {...} as const` might be less type-backed guarantee that `tabContentByType` contains keys for each and every `UnifiedSearchableContentType`.

With the explicit `Record<UnifiedSearchableContentType, ...>` type in place, in addition to ensuring a consistent shape between each value, we get type warnings if `tabContentByType` has missing properties or unknown properties:

| Missing property | Unknown property |
| - | - | 
| <img width="1025" alt="missing-property" src="https://github.com/hashicorp/dev-portal/assets/4624598/cd565064-7565-4277-8d85-dc10876fe5a4"> | <img width="961" alt="unknown-property" src="https://github.com/hashicorp/dev-portal/assets/4624598/b062ca6e-f42a-4b35-9f0f-d4a8684c6bea"> |

That being said, definitely open to revisiting this if there's another way to get this kind of check through more lightweight type annotations 🙇‍♂️ Let me know what you think!